src/main/java/org/opensearch/security/http/HTTPClientCertAuthenticator.java

```diff
@@ -54,66 +63,130 @@ public class HTTPClientCertAuthenticator implements HTTPAuthenticator {
-        this.settings = settings;
-        this.skipUsersMatcher = WildcardMatcher.from(settings.getAsList(OPENDISTRO_SECURITY_SSL_SKIP_USERS));
+    private enum AttributeType {
+        DN,
+        SAN
     }
 
-    @Override
-    public AuthCredentials extractCredentials(final SecurityRequest request, final ThreadContext threadContext) {
+    private record ParsedSAN(int type, @Nullable WildcardMatcher matcher) {
+    }
 
-        final String principal = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_SSL_PRINCIPAL);
+    private record ParsedAttribute(AttributeType type, String dnAttr, ParsedSAN san) {
+
+        static ParsedAttribute dn(String dnAttr) {
+            return new ParsedAttribute(AttributeType.DN, dnAttr, null);
+        }
+
+        static ParsedAttribute san(ParsedSAN san) {
+            return new ParsedAttribute(AttributeType.SAN, null, san);
+        }
+    }
 
-        if (!Strings.isNullOrEmpty(principal)) {
+    // Accept forms:
+    // "cn" -> DN:cn
+    // "dn:cn" -> DN:cn
+    // "san:EMAIL" -> SAN type EMAIL, no glob (match all of that SAN)
+    // "san:EMAIL:glob" -> SAN type EMAIL, glob
+    private ParsedAttribute parseAttributeSetting(String raw) {
+        if (Strings.isNullOrEmpty(raw)) return null; // “not configured”
+        final String s = raw.trim();
 
-            final String usernameAttribute = settings.get("username_attribute");
-            final String rolesAttribute = settings.get("roles_attribute");
+        // SAN form: san:TYPE[:glob]
+        if (s.regionMatches(true, 0, "san:", 0, 4)) {
+            final String rest = s.substring(4); // after "san:"
+            final int firstColon = rest.indexOf(':');
+            final String sanField = (firstColon >= 0) ? rest.substring(0, firstColon) : rest;
+            final String glob = (firstColon >= 0) ? rest.substring(firstColon + 1) : null;
 
-            if (skipUsersMatcher.test(principal)) {
-                log.debug("Skipped user client cert  authentication of user {} as its in skip_users list ", principal);
+            final SANType sanType;
+            try {
+                sanType = parseSanTypeToken(sanField);
+            } catch (IllegalArgumentException e) {
+                log.warn("Unsupported SAN type '{}' in attribute '{}'", sanField, raw);
                 return null;
             }
 
-            try {
-                final LdapName rfc2253dn = new LdapName(principal);
-                String username = principal.trim();
-                String[] backendRoles = null;
-
-                if (usernameAttribute != null && usernameAttribute.length() > 0) {
-                    final List<String> usernames = getDnAttribute(rfc2253dn, usernameAttribute);
-                    if (usernames.isEmpty() == false) {
-                        username = usernames.get(0);
-                    }
+            WildcardMatcher matcher = null; // null means pass-through
```

Comment:
nit: if you'd like to remove the special null handling, we can instantiate this as WildcardMatcher.ANY.

Reply:
done.
Thanks for the review @cwperks